src/libra/identifier/__init__.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) The Libra Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+"""LIP-5 Libra Account Identifier and Intent Identifier Utilities.
+
+See https://lip.libra.org/lip-5 for more details
+
+```python
+
+from libra import chain_ids, identifier
+
+# print hrp for premainnet by premainnet chain id
+print(identifier.HRPS[chain_ids.PREMAINNET.to_int()])
+```
+
+"""
+
+
+import typing
+from urllib import parse
+from typing import List
+
+from . import bech32
+from .. import libra_types, utils, chain_ids
+
+from .bech32 import bech32_address_encode, bech32_address_decode, Bech32Error
+from .subaddress import LIBRA_SUBADDRESS_SIZE, LIBRA_ZERO_SUBADDRESS, gen_subaddress
+
+LBR = "lbr"  # lbr for mainnet
+TLB = "tlb"  # tlb for testnet
+PLB = "plb"  # plb for premainnet
+
+HRPS: typing.Dict[int, str] = {
+    chain_ids.MAINNET.to_int(): LBR,
+    chain_ids.TESTNET.to_int(): TLB,
+    chain_ids.DEVNET.to_int(): TLB,
+    chain_ids.TESTING.to_int(): TLB,
+    chain_ids.PREMAINNET.to_int(): PLB,
+}
+
+
+class InvalidIntentIdentifierError(Exception):
+    pass
+
+
+class Intent:
+    """Intent is a struct hold data decoded from Libra Intent Identifier string"""
+
+    account_address: libra_types.AccountAddress
+    sub_address: typing.Optional[bytes]
+    currency_code: str
+    amount: int
+
+    def __init__(
+        self,
+        account_address: libra_types.AccountAddress,
+        sub_address: typing.Optional[bytes],
+        currency_code: str,
+        amount: int,
+    ) -> None:
+        self.account_address = account_address
+        self.sub_address = sub_address
+        self.currency_code = currency_code
+        self.amount = amount
+
+    @property
+    def account_address_bytes(self) -> bytes:
+        return self.account_address.to_bytes()
+
+
+def encode_intent(encoded_account_identifier: str, currency_code: str, amount: int) -> str:
```

Comment:
Why not include these methods in the Intent class?

Reply:
because it is easier do: identifier.encode_intent(...)

I think it's also fine to add encode method to Intent class, just think init an Intent class first then do encode seems not necessary, for example, when dump object to json.
another reason is Intent class has no hrp, although we can add it there.
but I'm not sure how it is used in production. 
the question is:  is it used when you have encoded account identifier (which includes account address, subaddress and hrp), or most likely you have account address, subaddress and hrp, then you want to create intent id.

current implementation assumes you have encoded account identifier in hand and tries to encode intent.
because I think you get encoded account identifier from somewhere else, for example scanning barcode, hence you don't really need to decode the account identifier first.

Reply:
I think this interface probably is not good.

In case of creating intent id:
1. you want others' pay you: VASP has account address, generate your subaddress, then create it, hence maybe use Intent is better.
2. you want pay others: in this case you received intent id, and need decode it for payment

maybe improve later.